Improve: Clean up keychain entries when deleting profiles and creating new ones

mpconley opened the pull request:
#### Brief overview of PR changes/additions

Adds keychain cleanup functionality to connection profile management:

- **Profile deletion cleanup**: When deleting profiles via the "Remove" button, now removes associated keychain entries ("character" and "proxy" passwords) in addition to removing profile directories
- **Orphaned entry detection**: When creating new profiles, detects and removes orphaned keychain entries from profiles previously deleted outside Mudlet interface  
- **Async implementation**: Uses existing CredentialManager async API with proper callback handling and debug logging
- **Helper method**: Refactored profile save logic into `continueProfileSave()` to support async orphaned entry cleanup

#### Motivation for adding to Mudlet

Prevents accumulation of orphaned credentials in system keychains (macOS Keychain, Windows Credential Store, Linux Secret Service) by ensuring complete profile lifecycle management. Previously, keychain entries were left behind when profiles were deleted, and orphaned entries could interfere with new profile creation.

#### Other info (issues closed, discussion etc)

Builds upon the secure credential management system from PR #7956. Follows the same async architecture patterns with non-blocking UI operations, proper memory management, and callback-based error handling.

Changes are only active when secure password storage is enabled.

---

Video with evidence of the keychain being cleared after a profile was created, closed, removed, recreated:

https://github.com/user-attachments/assets/a5e3e126-8b49-4871-8445-97b938a7c57e

---

Video with evidence of the keychain being cleared upon re-creating a profile that has a keychain entry in the past:

https://github.com/user-attachments/assets/15b86307-783d-49a4-87cc-06e28a5674b1

mpconley committed:
initial

vadi2 commented on src/dlgConnectionProfiles.cpp:
```diff
@@ -737,6 +803,34 @@ void dlgConnectionProfiles::reallyDeleteProfile(const QString& profile)
+                if (success) {
+                    qDebug() << "dlgConnectionProfiles: Successfully cleaned up character password for deleted profile" << profile;
+                } else {
+                    qDebug() << "dlgConnectionProfiles: Failed to clean up character password for deleted profile" << profile << ":" << errorMessage;
```
With the code working, do we still need the extensive debug checks? The rest of Mudlet (tries) to keep debug spam to a minimum.

mpconley replied:
Removed

vadi2 commented on src/dlgConnectionProfiles.cpp:
```diff
@@ -607,10 +607,76 @@ void dlgConnectionProfiles::slot_saveName()
+        auto* credManager = new CredentialManager(this);
+        credManager->retrieveCredential(newProfileName, "character", 
+            [this, credManager, newProfileName, pItem, newProfileHost, newProfilePort, newProfileSslTsl]
+            (bool foundCharacterEntry, const QString& characterPassword, const QString& errorMessage) {
+                Q_UNUSED(characterPassword)
+                Q_UNUSED(errorMessage)
```
Do characterPassword and errorMessage really need to be arguments if we discard them right away?

mpconley replied:
Good catch on the unused parameters. I kept the Q_UNUSED approach for these reasons:

1. API Contract: The CredentialManager::retrieveCredential() callback signature is fixed - we need all 3 parameters even though we only use the boolean result for existence checking.

2. Qt Convention: Q_UNUSED is the standard Qt pattern for required-but-unused callback parameters, consistent with the rest of the Mudlet codebase.

3. Documentation Value: Named parameters (characterPassword, errorMessage) document what the API provides, making the code self-documenting.

4. Adding a specialized credentialExists() method would add API complexity for this single use case. The current approach reuses existing, well-tested functionality.

The alternative of unnamed parameters (bool found, const QString&, const QString&) would be cleaner syntax but loses the self-documenting aspect and isn't typical Qt style.

Happy to change if you prefer a different approach, but this follows established Mudlet patterns.

vadi2 commented on src/dlgConnectionProfiles.cpp:
```diff
@@ -607,10 +607,76 @@ void dlgConnectionProfiles::slot_saveName()
+                                        if (success) {
+                                            qDebug() << "dlgConnectionProfiles: Successfully cleaned up orphaned character password for" << newProfileName;
+                                        } else {
+                                            qDebug() << "dlgConnectionProfiles: Failed to clean up orphaned character password for" << newProfileName << ":" << errorMessage;
```
Perhaps the debug logging can be removed now that the code has been worked out

mpconley replied:
Also removed

mpconley committed:
Merge branch 'development' into keychain_cleanup

mpconley committed:
Address reviewer feedback: reduce debug logging and keep Q_UNUSED

- Remove excessive debug logging, only log errors with qWarning()
- Keep Q_UNUSED approach for callback parameters per Qt conventions
- Maintains API documentation and follows Mudlet codebase patterns
